src/cli/bin.ts

```diff
@@ -3,47 +3,100 @@
 import { Command } from 'commander';
 import { default as getPath } from 'path';
 import { readFile } from 'fs/promises';
-import { parse, Source, ASTNodeKind, TypeDefinitionNode } from '../language';
+import { parse, Source, DocumentNode } from '../language';
 import { AnySpecSchema } from '../runtypes';
 import { validate, baseRules } from '../validation';
 import { AnySpecError, printError } from '../error';
 import { sync as glob } from 'globby';
 import ora from 'ora';
-
-function isEndpoint(val: string) {
-  return val.includes('.endpoints.');
-}
-
-function isModel(val: string) {
-  return val.includes('.models.');
-}
-
-function printCliError(error: string) {
-  return `${error}\n\n------------------------------------------------------------------------------------------\n`;
-}
+import { concatAST } from '../language/concatAST';
 
 async function main() {
   const program = new Command();
   program
-    .requiredOption('-o, --out-dir <dir>', 'path to a directory for a generated SDK')
-    .arguments('<specFiles...>');
+    .option('-o, --outDir <dir>', 'path to a directory for a generated openapi')
+    .option('-ns, --namespaces [namespaces...]', 'array of existed namespaces')
+    .option(
+      '-cns, --commonNamespace <commonNamespace>',
+      'name of common namespace where shared definitions stored',
+      'common',
+    )
+    .arguments('<specFiles>');
 
   program.parse();
 
   const { args } = program;
 
+  const options = program.opts() as {
+    commonNamespace: string;
+    namespaces?: string[];
+    outDir?: string;
+  };
+
+  const { namespaces, outDir, commonNamespace } = options;
+
+  if (!namespaces) {
+    throw new Error('please provide namespaces');
+  }
+
   const argPaths = args.map((arg) => getPath.resolve(process.cwd(), arg));
 
   const argumentPath = argPaths[0];
   const processingSpinner = ora(`Processing spec: ${argumentPath}`).start();
 
   const specFilePaths = glob(`${argumentPath}/**/*.tinyspec`);
 
+  const sources = await mapPathsToSources(specFilePaths);
+
+  const groupedSources = groupSourcesByNamespaces({ sources, commonNamespace, namespaces });
+
+  const { groupedParsedDocuments, parsingErrors } = getGroupedDocuments(
```

Comment:
Maybe it is a good case to use destructured object as a parameter here

smth like this
```
  const { groupedParsedDocuments, parsingErrors } = getGroupedDocuments({
    groupedSources,
    unexpectedErrorHandler: (error: Error) => {
      console.error('Unknown error during parsing', error);
      processingSpinner.fail();
      process.exit(1);
    },
  });
```

Reply:
2 arguments is pretty good for simple parameters